builtin/credential/ldap/backend.go

```diff
@@ -132,6 +132,14 @@ func (b *backend) Login(ctx context.Context, req *logical.Request, username stri
 		return nil, logical.ErrorResponse(err.Error()), nil, nil
 	}
 
+	if cfg.AnonymousGroupSearch {
+		c.Close()
```

Comment:
why this close here?

Reply:
I believe this is intended as a workaround for the issue described in https://github.com/hashicorp/vault/issues/6111.
From what I understand, when using Google's Secure LDAP, their implementation is not able to correctly handle a second bind request.

Reply:
Yes, you need to perform two requests with two different types of binds.

Reply:
More detailed response: you need to make an initial LDAP request with your username(email)/password along with the client certificates, in order to check if your login credentials are valid - but this bind will not return any groups if you attempt to look up the groups for that particular user.

To get group data for a user - for some reason it must be done with the client certificates under an anonymous bind.